src/Nethermind/Nethermind.Synchronization/FastSync/VerifyStateOnStateSyncFinished.cs

```diff
@@ -25,13 +25,18 @@ public class VerifyStateOnStateSyncFinished(
 {
     private readonly ILogger _logger = logManager.GetClassLogger<VerifyStateOnStateSyncFinished>();
 
+    bool _wasRun = false;
+
     public void Start()
     {
         treeSync.SyncCompleted += TreeSyncOnOnVerifyPostSyncCleanup;
     }
 
     private void TreeSyncOnOnVerifyPostSyncCleanup(object? sender, ITreeSync.SyncCompletedEventArgs evt)
     {
+        if (_wasRun) return;
+        _wasRun = true;
```

Comment:
Maybe better just to deregister from handler?
```suggestion
        treeSync.SyncCompleted -= TreeSyncOnOnVerifyPostSyncCleanup;
```